node/node.go

```diff
@@ -129,7 +130,8 @@ type Node struct {
 	err              error
 	agent            *agent.Agent
 	manager          *manager.Manager
-	notifyNodeChange chan *api.Node // used to send role updates from the dispatcher api on promotion/demotion
+	notifyNodeChange chan *agent.NodeChanges // used to send role updates from the dispatcher api on promotion/demotion
```

Comment:
The `nodeTLSChange` channel is so that the agent can report a TLS change to the dispatcher after the security config has changed.  The `notifyNodeChange` channel is for the dispatcher to notify the node that it needs to change something about it's TLS configuration (either renew the cert, because the node's status has changed, or replace its root certificate).